src/pycsou/opt/solver/cg.py

```diff
@@ -0,0 +1,132 @@
+import warnings
+
+import pycsou.abc.operator as pyco
+import pycsou.abc.solver as pycs
+import pycsou.opt.stop as pycos
+import pycsou.runtime as pycrt
+import pycsou.util as pycu
+import pycsou.util.ptype as pyct
+
+
+def is_broadcastable(arr1, arr2):
+    shp1, shp2 = arr1.shape, arr2.shape
+    if shp1[-1] != shp2[-1]:
+        return False
+    for a, b in zip(shp1[:-1], shp2[:-1]):
+        if a == 1 or b == 1 or a == b:
+            pass
+        else:
+            return False
+    return True
+
+
+class CG(pycs.Solver):
+    r"""
+    Conjugate Gradient Method.
+    The Conjugate Gradient method solves the system of linear equations of the form
+    .. math::
+       {\mathbf{A}\mathbf{x} = \mathbf{b}},
+    for the vector :math:`\mathcal{x}: \mathbb{x}^N`.
+    where:
+    * :math:`\mathcal{A}: \mathbb{R}^M\times^N` is a *symmetric* *positive definite* operator.
```

Comment:
You use both M and N as indices here, but A must be square.
Double-check the whole docstring and standardize on one letter only.

Also, rewrite this as `\mathcal{A}: \mathbb{R}^{N} \to \mathbb{R}^{N}`

Reply:
In `CG.fit()`, the docstring says "the minimization problem defined in :py:meth:`CG.__init__`", but Ax=b is not a minimization problem.
You should instead write \argmin_{x\in\bR^{N}} ||Ax - b||_{2}^{2}, where A is ...